modules/dataframe/src/properties/filterlistproperty.cpp

```diff
@@ -0,0 +1,207 @@
+            PropertySemantics::Text));
+        floatItem->addProperty(std::make_unique<FloatProperty>(
+            "epsilon", "Epsilon", 0.0f, 0.0f, std::numeric_limits<float>::max(), 0.1f,
+            InvalidationLevel::InvalidOutput, PropertySemantics::Text));
+
+        addPrefab(std::move(floatItem));
+    }
+    {
+        auto doubleItem =
+            std::make_unique<BoolCompositeProperty>("doubleItem", "Double Comparison", true);
+        doubleItem->addProperty(columnProp());
+        doubleItem->addProperty(std::make_unique<TemplateOptionProperty<filters::NumberComp>>(
+            "comp", "Comparison",
+            std::vector<OptionPropertyOption<filters::NumberComp>>{
+                {"equal", "equal (==)", filters::NumberComp::Equal},
+                {"notEqual", "Not Equal (!=)", filters::NumberComp::NotEqual},
+                {"less", "Less (<)", filters::NumberComp::Less},
+                {"lessEqual", "Less Equal (<=)", filters::NumberComp::LessEqual},
+                {"greater", "greater (>)", filters::NumberComp::Greater},
+                {"greaterEqual", "Greater Equal (>=)", filters::NumberComp::GreaterEqual}},
+            0));
+        doubleItem->addProperty(std::make_unique<DoubleProperty>(
+            "value", "Value", 0.0, std::numeric_limits<double>::lowest(),
+            std::numeric_limits<double>::max(), 0.1, InvalidationLevel::InvalidOutput,
+            PropertySemantics::Text));
+        doubleItem->addProperty(std::make_unique<DoubleProperty>(
+            "epsilon", "Epsilon", 0.0, 0.0, std::numeric_limits<double>::max(), 0.1,
+            InvalidationLevel::InvalidOutput, PropertySemantics::Text));
+
+        addPrefab(std::move(doubleItem));
+    }
+    {
+        auto intRange = std::make_unique<BoolCompositeProperty>("intRangeItem", "Int Range", true);
+        intRange->addProperty(columnProp());
+        intRange->addProperty(std::make_unique<IntMinMaxProperty>(
+            "range", "Integer Range", 0, 100, std::numeric_limits<int>::min(),
+            std::numeric_limits<int>::max(), 1, 0, InvalidationLevel::InvalidOutput,
+            PropertySemantics::Text));
+
+        addPrefab(std::move(intRange));
+    }
+    {
+        auto floatRange =
+            std::make_unique<BoolCompositeProperty>("floatRangeItem", "Float Range", true);
+        floatRange->addProperty(columnProp());
+        floatRange->addProperty(std::make_unique<FloatMinMaxProperty>(
+            "range", "Float Range", 0.0f, 100.0f, std::numeric_limits<float>::lowest(),
+            std::numeric_limits<float>::max(), 0.1f, 0.0f, InvalidationLevel::InvalidOutput,
+            PropertySemantics::Text));
+
+        addPrefab(std::move(floatRange));
+    }
+    {
+        auto doubleRange =
+            std::make_unique<BoolCompositeProperty>("doubleRangeItem", "Ddouble Range", true);
+        doubleRange->addProperty(columnProp());
+        doubleRange->addProperty(std::make_unique<DoubleMinMaxProperty>(
```

Comment:
Would it be possible to only have a Value Range filter of double type and use that for all?

I think that it would be better if a user did not have to care about the value type of the column.

Reply:
Same reasoning goes for the “match” case

Reply:
From a Filter perspective, the type of filter just determines the type a string is converted to before the comparison is applied.

Reply:
Added a setting to the ListProperty for enabling subsets of properties.